### ISSUE TITLE
Make code more reusable

### ISSUE THREAD

**Issue by impredicative** [5amrun/feature_selection_using_sensitivity_analysis#2]
Your paper is interesting. There is a shortage of feature selection implementations. Users would like to use your code if it was a lot more reusable, preferably also with the latest Python 3.7. I'm requesting that your code be made more reusable.

Right now I don't, for example, know where "SALib" is coming from. Using "%run fnn.py" and 
"%run import_data" is not professional either.

**Comment by 5amrun** [comment]
I agree. In fact, it was literally done in one day and was a side project. I'm planning to run it on Kaggle feature selection tasks. Yeah, the algorithm was surprisingly good! A nice assessment of various features.

**Comment by xingyu321** [comment]
Your code %fnn.py, Are some sub-functions missing？such as AttributeError: 'FnnSingleLayer' object has no attribute 'check_perform'。

**Comment by frdmu** [comment]
I have the same problem when running program: AttributeError: 'FnnSingleLayer' object has no attribute 'check_perform'
